File: app/src/main/java/com/close/hook/ads/hook/SettingsManager.java
package com.close.hook.ads.hook;

import com.close.hook.ads.hook.preference.PreferencesHelper;

public class SettingsManager {

    private final PreferencesHelper prefsHelper;
    private final String packageName;

    public SettingsManager(PreferencesHelper prefsHelper, String packageName) {
        this.prefsHelper = prefsHelper;
        this.packageName = packageName;
    }

    public boolean isHandlePlatformAdEnabled() {
        return prefsHelper.getBoolean("switch_one_" + packageName, false);
    }

    public boolean isRequestHookEnabled() {
        return prefsHelper.getBoolean("switch_two_" + packageName, false);
    }

    public boolean isHideVPNStatusEnabled() {
        return prefsHelper.getBoolean("switch_three_" + packageName, false);
    }

    public boolean isDisableFlagSecureEnabled() {
        return prefsHelper.getBoolean("switch_four_" + packageName, false);
    }

    public boolean isDisableShakeAdEnabled() {
        return prefsHelper.getBoolean("switch_five_" + packageName, false);
    }

    public boolean isHideEnivEnabled() {
        return prefsHelper.getBoolean("switch_six_" + packageName, false);
    }

    public boolean isDisableClipboard() {
        return prefsHelper.getBoolean("switch_seven_" + packageName, false);
    }
}

File: app/src/main/java/com/close/hook/ads/hook/SettingsManager.kt
package com.close.hook.ads.hook

import com.close.hook.ads.preference.HookPrefs

class SettingsManager(
    private val packageName: String,
    private val prefsHelper: HookPrefs
) {

    private fun key(suffix: String): String = "switch_${suffix}_$packageName"

    val isHandlePlatformAdEnabled: Boolean
        get() = prefsHelper.getBoolean(key("one"), false)

    val isRequestHookEnabled: Boolean
        get() = prefsHelper.getBoolean(key("two"), false)

    val isHideVPNStatusEnabled: Boolean
        get() = prefsHelper.getBoolean(key("three"), false)

    val isDisableFlagSecureEnabled: Boolean
        get() = prefsHelper.getBoolean(key("four"), false)

    val isDisableShakeAdEnabled: Boolean
        get() = prefsHelper.getBoolean(key("five"), false)

    val isHideEnivEnabled: Boolean
        get() = prefsHelper.getBoolean(key("six"), false)

    val isDisableClipboard: Boolean
        get() = prefsHelper.getBoolean(key("seven"), false)

    val isHookTipEnabled: Boolean
        get() = prefsHelper.getBoolean(key("eight"), false)
}
